src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperBlockRewriter.cs

```diff
@@ -119,7 +122,7 @@ public static class TagHelperBlockRewriter
                 else if (name == null && symbol.Type == HtmlSymbolType.Text)
                 {
                     name = symbol.Content;
-                    attributeValueStartLocation = SourceLocation.Advance(span.Start, name);
+                    attributeValueStartLocation = SourceLocation.Advance(attributeValueStartLocation, name);
                 }
                 else if (symbol.Type == HtmlSymbolType.Equals)
                 {
```

Comment:
right, sorry; it's just whitespace after equals, missing name, and missing value cases that could trip this code up.  and at least some of those (whitespace) won't reach here due to parser issues.  know missing value _does_ lead to code gen.  what about missing name?